### PR TITLE
Parameterize Github URL at infrastructure setup

### PR DESCRIPTION
**Pull request by YasminBeatriz**
Instead of having GitHub host name (github.com) hard-coded at the setup, now replace it with `$GIT_REPOS_BASE_URL` so it allows users to use enterprise instances of GitHub as well (github.ibm.com, for example).

### CODE REVIEW

File: ansible/roles/builds/templates/github_ssh_config.j2 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-host github.com
- HostName github.com
+host github
+ HostName params.GITHUB_REPOS_BASE_URL
```
Comment (farosas) [inline ansible/roles/builds/templates/github_ssh_config.j2]: This doesn't look like it would work, am I missing something? `params` is a Jenkins variable isn't it?

File: pipeline/build/stages.groovy [hunk 2]
```diff
@@ -47,7 +47,7 @@ def authorize() {
   } else {
     echo("Retrieving authorized users from GitHub organization " +
          params.GITHUB_ORGANIZATION_NAME)
-    String orgMembersURL = ("https://api.github.com/orgs/" +
+    String orgMembersURL = ("https://api.$params.GITHUB_REPOS_BASE_URL/orgs/" +
```
Comment (farosas) [inline pipeline/build/stages.groovy]: Please be consistent in the variable access syntax.

File: ansible/vars-bvt.yaml [hunk 3]
```diff
@@ -12,6 +12,6 @@ avocado_conf: "{{avocado_conf_dir}}/avocado.conf"
 test_providers_dir: "{{avocado_data_dir}}/avocado-vt/test-providers.d/downloads"
 
 avocado_repos:
-  - { repo: "https://github.com/autotest/tp-libvirt.git", dest: "{{test_providers_dir}}/io-github-autotest-libvirt" }
-  - { repo: "https://github.com/autotest/tp-qemu.git", dest: "{{test_providers_dir}}/io-github-autotest-qemu" }
-  - { repo: "https://github.com/spiceqa/tp-spice.git", dest: "{{test_providers_dir}}/io-github-spiceqa-spice" }
+  - { repo: "https://{{pipeline_constants.GITHUB_REPOS_BASE_URL}}/autotest/tp-libvirt.git", dest: "{{test_providers_dir}}/io-github-autotest-libvirt" }
```
Comment (farosas) [inline ansible/vars-bvt.yaml]: This is a bit confusing. "base url" looks like it would contain the _https://_ part of the URL too. What about `GITHUB_DOMAIN`?

**Review by farosas — CHANGES_REQUESTED** [review]

File: pipeline/build/parameters.groovy [hunk 4]
```diff
@@ -1,4 +1,7 @@
 Map pipelineParameters = [
+  GITHUB_DOMAIN: [
+    defaultValue: constants.GITHUB_DOMAIN,
+    description: 'GitHub instance.'],
```
Comment (farosas) [inline pipeline/build/parameters.groovy]: "Domain of the GitHub instance (github.com, github.org.com)"

**Review by farosas — CHANGES_REQUESTED** [review]

File: ansible/vars-bvt.yaml [hunk 5]
```diff
@@ -12,6 +12,6 @@ avocado_conf: "{{avocado_conf_dir}}/avocado.conf"
 test_providers_dir: "{{avocado_data_dir}}/avocado-vt/test-providers.d/downloads"
 
 avocado_repos:
-  - { repo: "https://github.com/autotest/tp-libvirt.git", dest: "{{test_providers_dir}}/io-github-autotest-libvirt" }
-  - { repo: "https://github.com/autotest/tp-qemu.git", dest: "{{test_providers_dir}}/io-github-autotest-qemu" }
-  - { repo: "https://github.com/spiceqa/tp-spice.git", dest: "{{test_providers_dir}}/io-github-spiceqa-spice" }
+  - { repo: "https://{{pipeline_constants.GITHUB_DOMAIN}}/autotest/tp-libvirt.git", dest: "{{test_providers_dir}}/io-github-autotest-libvirt" }
+  - { repo: "https://{{pipeline_constants.GITHUB_DOMAIN}}/autotest/tp-qemu.git", dest: "{{test_providers_dir}}/io-github-autotest-qemu" }
+  - { repo: "https://{{pipeline_constants.GITHUB_DOMAIN}}/spiceqa/tp-spice.git", dest: "{{test_providers_dir}}/io-github-spiceqa-spice" }
```
Comment (olavph) [inline ansible/vars-bvt.yaml]: Those are upstream projects and repositories that we do not modify. They should keep the fixed "github.com" domain.

**Review by olavph — CHANGES_REQUESTED** [review]

**Review by farosas — APPROVED** [review]

**Review by olavph — APPROVED** [review]

File: ansible/vars.yaml [hunk 6]
```diff
@@ -1,5 +1,6 @@
 ---
 pipeline_constants:
+  GITHUB_DOMAIN:
```
Comment (olavph) [inline ansible/vars.yaml]: This should have a default value "github.com"

File: ansible/roles/jenkins-seed-job/templates/seed_job_config.xml.j2 [hunk 7]
```diff
@@ -13,7 +13,7 @@
         <hudson.model.StringParameterDefinition>
           <name>REPOSITORY_URL</name>
           <description>Repository to download the DSL job descriptors.</description>
-          <defaultValue>https://github.com/{{pipeline_constants.GITHUB_ORGANIZATION_NAME}}/infrastructure.git</defaultValue>
+          <defaultValue>https://{{pipeline_constants.GITHUB_DOMAIN}}/{{pipeline_constants.GITHUB_ORGANIZATION_NAME}}/infrastructure.git</defaultValue>
         </hudson.model.StringParameterDefinition>
```
Comment (olavph) [inline ansible/roles/jenkins-seed-job/templates/seed_job_config.xml.j2]: GITHUB_DOMAIN needs to be added as a parameter in this file. You can place it right before GITHUB_ORGANIZATION_NAME

**Review by olavph — CHANGES_REQUESTED** [review]

File: ansible/roles/jenkins-seed-job/templates/seed_job_config.xml.j2 (hunk 7 above)
Comment (olavph) [reply ansible/roles/jenkins-seed-job/templates/seed_job_config.xml.j2]: The SSH key is not being used because the host `github.com` no longer matches the entry `github` from the config file. You'll have to replace every occurrence of `ssh://git@github.com/` to `ssh://git@github/`.
I'd also change every HTTPS URL to SSH, to make use of the same configuration and detect authorization issues earlier. With this, you can also remove my suggestion above and the variable `{{pipeline_constants.GITHUB_DOMAIN}}` from this file.

**Review by farosas — CHANGES_REQUESTED** [review]
Let's avoid overwriting the user's `.ssh/config` file because the user might be a real one.

File: jenkins_jobs/build_old_host_os_versions.groovy [hunk 8]
```diff
@@ -5,7 +5,7 @@ job('build_old_host_os_versions') {
 		"${GITHUB_ORGANIZATION_NAME}",
 		'GitHub organization from where the Host OS repositories will be checked out.')
     stringParam('BUILDS_REPO_URL',
-		"https://${GITHUB_DOMAIN}/${GITHUB_ORGANIZATION_NAME}/builds.git",
+		"ssh://git@github.com/${GITHUB_ORGANIZATION_NAME}/builds.git",
```
Comment (olavph) [inline jenkins_jobs/build_old_host_os_versions.groovy]: Remove `.com`

File: jenkins_jobs/build_old_host_os_versions.groovy [hunk 9]
```diff
@@ -5,7 +5,7 @@ job('build_old_host_os_versions') {
 		"${GITHUB_ORGANIZATION_NAME}",
 		'GitHub organization from where the Host OS repositories will be checked out.')
     stringParam('BUILDS_REPO_URL',
-		"https://github.com/${GITHUB_ORGANIZATION_NAME}/builds.git",
+		"https://${GITHUB_DOMAIN}/${GITHUB_ORGANIZATION_NAME}/builds.git",
```
Comment (olavph) [inline jenkins_jobs/build_old_host_os_versions.groovy]: Changing to `${GITHUB_DOMAIN}` (github.com) breaks the build because the SSH configuration for github (whose name change to `github` only) is not used. The next commit should be squashed into this one, so we don't have broken/unnecessary changes polluting the git history.
(This comment should be in an SSH URL, sorry)

**Review by olavph — CHANGES_REQUESTED** [review]

**Review by farosas — APPROVED** [review]

File: jenkins_jobs/build_old_host_os_versions.groovy (hunk 9 above)
Comment (olavph) [reply jenkins_jobs/build_old_host_os_versions.groovy]: The variable GITHUB_DOMAIN does not exist here. It would need to be created in the seed job to be accessible.
Since we are changing the HTTP URLs to SSH, this would be a useless change, that would need to be reverted in the next commit. It makes more sense to reorder the commits so that the change from HTTP to SSH comes first (still using the fixed github.com domain) and this commit that parameterizes the GitHub URL comes second.

By doing this, the **only** place where GITHUB_DOMAIN should appear is in the SSH config file.

**Review by olavph — CHANGES_REQUESTED** [review]

**Review by olavph — APPROVED** [review]
